File: nodes/subtract/subtract.process.js
export default async ({inputs, settings, config}) => {

  const a = Number(inputs.a) || 0;
  const b = Number(inputs.b) || 0;


  // try to handle floating point subtraction
  let result = a - b;

  result = Number(result.toFixed(10));
  
  return {
    result: result
  };
}; 

File: nodes/subtract/subtract.process.py
async def process(inputs, settings, config, nodeConfig):
  a = float(inputs.get("a", 0))
  b = float(inputs.get("b", 0))
  
  # Calculate the result
  result = a - b
  
  # Handle floating point precision issues
  # Convert to string with fixed precision and back to float
  result = float(format(result, '.10f'))
  
  return {
    "result": result
  } 